src/search/srpso.c

```
@@ -0,0 +1,239 @@
+    v += c1 * r1 * (s->pbest[i][j] - s->pos[i][j]);
+    v += c2 * r2 * (s->gbest[j] - s->pos[i][j]);
+
+    if (v > v_max) v = v_max;
+    if (v < -v_max) v = -v_max;
+
+    s->vel[i][j] = v;
+    s->pos[i][j] += v;
+  }
+}
+
+int sr_pso_optimize(const sr_pso_cfg *cfg, int ndim, real (*func)(real *),
```

Clang (14.0.6)

[warning] function 'sr_pso_optimize' has cognitive complexity of 28 (threshold 25)

(readability-function-cognitive-complexity)